### PR TITLE
clean up interface, add tests, fix makefile clean

### PR DESCRIPTION
**Pull request by ixje**
1) I cleaned up the interface to be more natural for python users. This means removing `.add()`, `.sub()` etc methods and only keep the dunder methods to support `b1 + b2` and alike 
2) added support for other common calls like `hex()`, `float()`, `int()`, `abs()` etc
3) changed logic to allow subclassing
4) added unittest tests in Python style (copied from neo-python)
5) added some (class) methods to give a similar interface to the default builtin `int` type. This will make it easier to drop it in as a replacement
6) fixed directory casing in Makefile for `make clean` command

### CODE REVIEW

**Comment by ixje** [comment]
ps: also fixed an issue where creating a BigInteger from a `bytearray` internally differed from creating it from a `str`. Equality would fail because the `_data` buffer for `bytearray` would be exact length, whereas for a string would be `stdsize` (=256).

**Comment by igormcoelho** [comment]
Now it looks like a professional BigInt class xD

File: csbiginteger/BigInteger.py [hunk 1]
```diff
@@ -67,56 +63,135 @@
...
         if self.__gt__(2**31-1):
             raise OverflowError("overflow on signed int32")
         if self.__lt__(-2**31):
             raise OverflowError("underflow on signed int32")
         return int(str(self))
 
-    def to_long(self):
+    def __float__(self) -> float:
+        """
+        Convert value to `long` type.
```
Comment (igormcoelho) [inline csbiginteger/BigInteger.py:123]: @ixje I didn't get this mention to `long` here... can you check?

File: csbiginteger/BigInteger.py [hunk 2]
```diff
@@ -67,56 +63,135 @@
...
+
+        Raises:
+            OverflowError: if value does not fit in a `long` according to the C# rules
+
+        Returns:
+             long: value in signed int32 limit
```
Comment (igormcoelho) [inline csbiginteger/BigInteger.py:129]: This one is strange to... in a __float__ context... mentions long, and also a int32...

File: csbiginteger/BigInteger.py [hunk 3]
```diff
@@ -67,56 +63,135 @@
...
+
+            sz = csbiginteger_lib.csbiginteger_init_s(strdata, base, self._data, self.std_size)
             if sz == 0:
                 raise ValueError('Something wrong with BigInteger. Zero size.')
             self._length = sz
 
+            # trim data to exact length
+            alloc = (ctypes.c_uint8 * self._length)
+            new_data = alloc.from_buffer(bytearray(self._data)[:self._length])
+            self._data = new_data
```
Comment (igormcoelho) [inline csbiginteger/BigInteger.py:101]: A very good change!! only occupies necessary space!

**Pull request merged**